src/providers/palm/chatComplete.ts

```diff
@@ -0,0 +1,81 @@
+import { Params } from "../../types/requestBody";
+import { PalmChatCompleteResponse } from "../../types/responseBody";
+import { ChatCompletionResponse, ErrorResponse, ProviderConfig } from "../types";
+
+// TODOS: this configuration does not enforce the maximum token limit for the input parameter. If you want to enforce this, you might need to add a custom validation function or a max property to the ParameterConfig interface, and then use it in the input configuration. However, this might be complex because the token count is not a simple length check, but depends on the specific tokenization method used by the model.
+
+export const PalmChatCompleteConfig: ProviderConfig = {
+    model: {
+        param: "model",
+        required: true,
+        default: "gpt-3.5-turbo",
+    },
+    messages: {
+        param: "prompt",
+        default: "",
+        transform: (params: Params) => {
+            const { examples, context, messages } = params
+            const palm_messages = messages?.map(message => ({
+                "author": message.role,
+                "content": message.content
+            }))
+            const prompt = {
+                messages: palm_messages,
+                examples,
+                context
+            }
+            return prompt
+        }
+    },
+    temperature: {
+        param: "temperature",
+        default: 1,
+        min: 0,
+        max: 1,
+    },
+    top_p: {
+        param: "topP",
+        default: 1,
+        min: 0,
+        max: 1,
+    },
+    top_k: {
+        param: "topK",
+        default: 1,
+        min: 0,
+        max: 1,
+    },
+    candidate_count: {
```

Comment:
candidate_count seems like an equivalent to n which is there in existing specification. Please reuse that and make the transformation to palm specific key